.github/workflows/main.yml

```diff
@@ -59,3 +66,23 @@ jobs:
               name: playwright-artifacts
               path: frontend/tests/playwright/test-results/artifacts
               retention-days: 7
+
+         - name: Prepare results to upload for project matrix
+           if: success() || failure()
+           id: test-result
+           run: |
+              MATRIX_NAME=$(echo "${{ matrix.project }}" | tr '[:upper:]' '[:lower:]' | tr ' ' '-')
+              RESULT_ORIGINAL_NAME="playwright-result-$BUILD_COMMIT"
+              RESULT_FINAL_NAME="$RESULT_ORIGINAL_NAME-$MATRIX_NAME"
+              mv "frontend/$RESULT_ORIGINAL_NAME" "frontend/$RESULT_FINAL_NAME"
+              RESULT_PATH="frontend/$RESULT_FINAL_NAME"
+              echo "Result path: $RESULT_PATH"
+              echo "result-path=$RESULT_PATH" >> $GITHUB_OUTPUT
+
+         - name: Upload Playwright Test Result Metrics
+           if: success() || failure()
```

Comment:
Why not `always()`? (Same as above)

Reply:
If the job is canceled/timed out when the `Run Playwright Tests` step is running, then it doesn't generate the test results file. 

Since, the `always()` includes `cancelled()` and `skipped()`, I opted to use `success() || failure()` because we want to upload the test results on either test fails or passes.